etc/compile_check.sh

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env bash
+set -e # exit when any command fails
+set -x # show all commands being run
+
+GC=go
+COMPILE_CHECK_DIR="internal/test/compilecheck"
+DEV_MIN_VERSION=1.19
+
+# version will flatten a version string of upto 4 components for inequality
+# comparison.
+function version {
+	echo "$@" | awk -F. '{ printf("%d%03d%03d%03d\n", $1,$2,$3,$4); }';
+}
+
+# dev_compile_check will attempt to build in the development environment. This
+# check will only run on environments where the Go version is greater than or
+# equal to the DEV_MIN_VERSION.
+function dev_compile_check {
```

Comment:
Optional: The `dev_compile_check` now seems redundant with the `build` Makefile target. Consider removing it.